egs/librispeech/ASR/pruned_transducer_stateless5/ConvRNN-T.py

```diff
@@ -0,0 +1,1029 @@
+                    warmup=warmup,
+                ) 
+                for i in range(num_layers)
+            ]
+        )
+        self.num_layers = num_layers
+
+        assert len(set(aux_layers)) == len(aux_layers)
+
+        assert num_layers - 1 not in aux_layers
+        self.aux_layers = aux_layers + [num_layers - 1]
+
+        self.combiner = RandomCombine(
+            num_inputs=len(self.aux_layers),
+            final_weight=0.5,
+            pure_prob=0.333,
+            stddev=2.0,
+        )
+
+    def forward(
+        self,
+        src: Tensor,
+        mask: Optional[Tensor] = None,
+        src_key_padding_mask: Optional[Tensor] = None,
+        warmup: float = 1.0,
+    ) -> Tensor:
+        r"""Pass the input through the encoder layers in turn.
+
+        Args:
+            src: the sequence to the encoder (required).
+            mask: the mask for the src sequence (optional).
+            src_key_padding_mask: the mask for the src keys per batch (optional).
+
+        Shape:
+            src: (S, N, E).
+            mask: (S, S).
+            src_key_padding_mask: (N, S).
+            S is the source sequence length, T is the target sequence length, N is the batch size, E is the feature number
+
+        """
+        output = src
+
+        outputs = []
+
+        for i, mod in enumerate(self.layers):
+            output = mod(
+                output,
+                src_key_padding_mask=src_key_padding_mask,
+            )
+            if i in self.aux_layers:
+                outputs.append(output)
+
+        output = self.combiner(outputs)
+
+        return output
+
+
+class SEModule(nn.Module):
+    """ SE Module as defined in original SE-Nets with a few additions
+    Additions include:
+        * divisor can be specified to keep channels % div == 0 (default: 8)
+        * reduction channels can be specified directly by arg (if rd_channels is set)
+        * reduction channels can be specified by float rd_ratio (default: 1/16)
+        * global max pooling can be added to the squeeze aggregation
+        * customizable activation, normalization, and gate layer
+    Modified from https://github.com/rwightman/pytorch-image-models/blob/main/timm/models/layers/squeeze_excite.py
+    """
+    def __init__(
+            self, channels, rd_ratio=1. / 16, rd_channels=None, rd_divisor=8, add_maxpool=False,
+            bias=True, act_layer=nn.ReLU, norm_layer=None, gate_layer='Sigmoid'):
+        super(SEModule, self).__init__()
+        self.add_maxpool = add_maxpool
+        if not rd_channels:
+            rd_channels = self.make_divisible(channels * rd_ratio, rd_divisor, round_limit=0.)
+        self.fc1 = nn.Conv1d(channels, rd_channels, kernel_size=1, bias=bias)
+        self.act = nn.ReLU(inplace=True)
+        self.fc2 = nn.Conv1d(rd_channels, channels, kernel_size=1, bias=bias)
+        self.gate = nn.Sigmoid()
+
+    def forward(self, x):
+        # x: (N, C, T)
+        x_se = x.mean(2, keepdim=True)
```

Comment:
`x.mean(2, keepdim=True)` here needs to consider padding ?

Reply:
You are right, I will fix it.

Reply:
You are right, I will fix it.

Reply:
Another question, if mean over all time steps T, is it suitable for streaming ?